src/librustc_mir/interpret/memory.rs

```diff
@@ -791,6 +791,21 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
         self.get_raw(ptr.alloc_id)?.read_c_str(self, ptr)
     }
 
+    /// Reads a 0x0000-terminated sequence of bytes from memory. Returns them as a slice.
+    /// Needed for reading wide-strings in Windows-OS
+    pub fn read_wide_str(&self, ptr: Scalar<M::PointerTag>) -> InterpResult<'tcx, &[u8]> {
+        let widestr_u8_initbyte = self.read_bytes(ptr, Size::from_bytes(1))?;
+        let mut widestr_len = 0; // length in bytes
+        unsafe {
```

Comment:
Could a safety comment be added to this block?

Reply:
Better yet, could you please not use `unsafe`? Just follow the implementation strategy of `read_c_str`, that one does not use unsafe, either.